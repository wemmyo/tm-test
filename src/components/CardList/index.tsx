import * as React from "react";
import { CardI } from "../../types";
import Card from "../Card";

interface CardListProps {
  cards: CardI[];
}

const CardList: React.FC<CardListProps> = ({ cards }) => {
  return (
    <div>
      {cards.map((card: CardI) => (
        <Card key={card.id} data={card} />
      ))}
    </div>
  );
};

export default CardList;
