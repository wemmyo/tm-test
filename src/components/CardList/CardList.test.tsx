import React from "react";
import { shallow } from "enzyme";
import CardList from "./index";

const mockCards = [
  {
    id: 1,
    label: "Student Life",
    apr: 0.189,
    balanceTransferOfferDuration: 0,
    purchaseOfferDuration: 6,
    credit: 1200,
  },
  {
    id: 2,
    label: "Anywhere Card",
    apr: 0.339,
    balanceTransferOfferDuration: 0,
    purchaseOfferDuration: 0,
    credit: 300,
  },
  {
    id: 3,
    label: "Liquid Card",
    apr: 0.339,
    balanceTransferOfferDuration: 12,
    purchaseOfferDuration: 6,
    credit: 3000,
  },
];

it("renders without crashing", () => {
  expect(shallow(<CardList cards={mockCards} />)).toMatchSnapshot();
});
