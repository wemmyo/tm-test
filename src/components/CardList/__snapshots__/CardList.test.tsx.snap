// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <Card
    data={
      Object {
        "apr": 0.189,
        "balanceTransferOfferDuration": 0,
        "credit": 1200,
        "id": 1,
        "label": "Student Life",
        "purchaseOfferDuration": 6,
      }
    }
    key="1"
  />
  <Card
    data={
      Object {
        "apr": 0.339,
        "balanceTransferOfferDuration": 0,
        "credit": 300,
        "id": 2,
        "label": "Anywhere Card",
        "purchaseOfferDuration": 0,
      }
    }
    key="2"
  />
  <Card
    data={
      Object {
        "apr": 0.339,
        "balanceTransferOfferDuration": 12,
        "credit": 3000,
        "id": 3,
        "label": "Liquid Card",
        "purchaseOfferDuration": 6,
      }
    }
    key="3"
  />
</div>
`;
